src/actions.js

```diff
@@ -135,50 +139,123 @@ export const clearInitWallet = () => ({ type: types.SET_INIT_WALLET, payload: nu
-  wallet.sendTransaction(address, amount, token, { pinCode: pin })
+  wallet.sendTransactionEvents(address, amount, token, { pinCode: pin })
 );
 
+/**
+ * Map history element to expected TxHistory model object
+ *
+ * element {Object} Tx history element with {txId, timestamp, balance, voided?}
+ * token {string} Token uid
+ */
+const mapTokenHistory = (element, token) => {
+  const data = {
+    txId: element.txId,
+    timestamp: element.timestamp,
+    balance: element.balance,
+    // in wallet service this comes as 0/1 and in the full node comes with true/false
+    voided: Boolean(element.voided),
+    tokenUid: token
+  };
+  return new TxHistory(data);
+};
+
+/**
+ * Get all tokens that this wallet has any transaction and fetch balance/history for each of them
+ * We could do a lazy history load only when the user selects to see the token
+ * but this would change the behaviour of the wallet and was not the goal of this moment
+ * We should do this in the future anwyay
+ *
+ * wallet {HathorWallet | HathorWalletServiceWallet} wallet object
+ */
+export const fetchHistoryAndBalance = async (wallet) => {
+  // First we get the tokens in the wallet
+  const tokens = await wallet.getTokens();
+
+  const tokensHistory = {};
+  const tokensBalance = {};
+  for (const token of tokens) {
+    /* eslint-disable no-await-in-loop */
+    const balance = await wallet.getBalance(token);
+    const tokenBalance = balance[0].balance;
+    tokensBalance[token] = { available: tokenBalance.unlocked, locked: tokenBalance.locked };
+    const history = await wallet.getTxHistory({ token_id: token });
+    tokensHistory[token] = history.map((element) => mapTokenHistory(element, token));
+    /* eslint-enable no-await-in-loop */
+  }
+
+  return { tokensHistory, tokensBalance };
+};
+
+/**
+ * Fetch pagination history for specific token
+ *
+ * wallet {HathorWallet | HathorWalletServiceWallet} wallet object
+ * token {string} Token uid
+ * history {Array} current token history array
+ */
+export const fetchMoreHistory = async (wallet, token, history) => {
+  const newHistory = await wallet.getTxHistory({ token_id: token, skip: history.length });
+  const newHistoryObjects = newHistory.map((element) => mapTokenHistory(element, token));
+
+  return newHistoryObjects;
+};
+
 export const startWallet = (words, pin) => (dispatch) => {
   // If we've lost redux data, we could not properly stop the wallet object
   // then we don't know if we've cleaned up the wallet data in the storage
   walletUtil.cleanLoadedData();
 
-  const connection = new Connection({
-    network: 'mainnet', // app currently connects only to mainnet
-    servers: ['https://mobile.wallet.hathor.network/v1a/'],
-  });
+  const networkName = 'testnet';
 
-  const beforeReloadCallback = () => {
-    dispatch(activateFetchHistory());
-  };
+  let wallet;
+  // TODO substitute for feature flag when rollout code is implemented
+  if (false) {
```

Comment:
Maybe create a variable `const useWalletService = false`, so we know what this condition is related to.

Reply:
Created a redux variable that is being used now as feature flag. https://github.com/HathorNetwork/hathor-wallet-mobile/pull/111/commits/6f6c05de5d4714040ee97ffdd2dcddf37e9ac619

Reply:
Great! Thanks!